app/src/androidTest/java/co/nimblehq/compose/crypto/ui/screen/detail/DetailScreenUiTest.kt

```diff
@@ -0,0 +1,160 @@
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(TestTagDetailLoading).assertIsDisplayed()
+        }
+    }
+
+    @Test
+    fun when_navigating_to_detail_screen_it_renders_chart_interval_buttons_properly() {
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flowOf(coinDetail)
+        every { mockGetCoinPricesUseCase.execute(any()) } returns flowOf(coinPrices)
+
+        with(composeAndroidTestRule) {
+            onNodeWithText(TimeIntervals.ONE_DAY.text).assertIsDisplayed()
+            onNodeWithText(TimeIntervals.ONE_WEEK.text).assertIsDisplayed()
+            onNodeWithText(TimeIntervals.ONE_MONTH.text).assertIsDisplayed()
+            onNodeWithText(TimeIntervals.ONE_YEAR.text).assertIsDisplayed()
+            onNodeWithText(TimeIntervals.FIVE_YEAR.text).assertIsDisplayed()
+        }
+    }
+
+    @Test
+    fun when_navigating_to_detail_screen_it_render_currentPrice_and_priceChangePercentage24hInCurrency_properly() {
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flowOf(coinDetail)
+        every { mockGetCoinPricesUseCase.execute(any()) } returns flowOf(coinPrices)
+
+        with(composeAndroidTestRule) {
+            coinDetail.marketData?.let { marketData ->
+                val currentPrice = "$${marketData.currentPrice[FIAT_CURRENCY]?.toFormattedString()}"
+                onNodeWithText(currentPrice).assertIsDisplayed()
+
+                val priceChangePercentage24hInCurrency = this.activity.getString(
+                    R.string.coin_profit_percent,
+                    abs(marketData.priceChangePercentage24hInCurrency[FIAT_CURRENCY] ?: 0.0).toFormattedString()
+                )
+                onNodeWithText(priceChangePercentage24hInCurrency).assertIsDisplayed()
+            }
+        }
+    }
+
+    @Test
+    fun when_navigating_to_detail_screen_it_render_coin_info_properly() {
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flowOf(coinDetail)
+        every { mockGetCoinPricesUseCase.execute(any()) } returns flowOf(coinPrices)
+
+        with(composeAndroidTestRule) {
+            coinDetail.marketData?.let { marketData ->
+                val marketCap = "$${marketData.marketCap[FIAT_CURRENCY]?.toFormattedString()}"
+                onNodeWithText(marketCap).assertIsDisplayed()
+
+                val allTimeHigh = "$${marketData.ath[FIAT_CURRENCY]?.toFormattedString()}"
+                onNodeWithText(allTimeHigh).assertIsDisplayed()
+
+                val allTimeLow = "$${marketData.atl[FIAT_CURRENCY]?.toFormattedString()}"
+                onNodeWithText(allTimeLow).assertIsDisplayed()
+            }
+        }
+    }
+
+    @Test
+    fun when_navigating_to_detail_screen_and_has_api_error_coin_price_chart_is_not_displayed() {
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flow {
+            throw Throwable(errorGeneric)
+        }
+        every { mockGetCoinPricesUseCase.execute(any()) } returns flowOf(coinPrices)
```

Comment:
same, mocking success response for `mockGetCoinPricesUseCase` could be moved to the `setUp` block.

```suggestion
        every { mockGetCoinDetailUseCase.execute(any()) } returns flow {
            throw Throwable(errorGeneric)
        }
```